### DISCUSSION TITLE
Response Status: 504 - Error: Gateway Time-out

### DISCUSSION THREAD

**Discussion by alipiry** [vercel/community#5524] [Help]
### Summary

Hi there,

I have this NextJS app router project hosting on Vercel and getting dynamic data using GraphQL from WordPress on WPEngine. When I push a new change, Vercel cannot build the project successfully and returns 504 Gateway Time-out error every time. Do you have any idea how to fix this?

### Steps to Reproduce

I tried changing the staticPageGenerationTimeout to something other than 60, but still, the issue occurred.

**Comment by alipiry** [comment]
@amyegan 🙏

**Reply by amyegan** [reply]
Do you run into this timeout error with other data sources, or only with that WordPress instance? It could be timing out because of missing or incorrect [environment variables](https://vercel.com/docs/projects/environment-variables). Please double check that they're all correct in project settings. 🙏

**Reply by amyegan** [reply]
Additional troubleshooting ideas: https://vercel.com/docs/errors/FUNCTION_INVOCATION_TIMEOUT

**Reply by alipiry** [reply]
Hey @amyegan I get this error only with the WP instance - ENVs are correct as it's been working for the past few weeks - When we updated some minor UI issues, this error occurred while generating static pages on the build!
